### ISSUE TITLE
HTTPRequest retrieves corrupted file with error

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#7345]
Tested on Godot 2.1.1, Windows 10 64 bits

I wanted to test the HTTPRequest node to download an image, with the following script:

```gdscript
extends Button


onready var _request = get_node("HTTPRequest")


func _ready():
    connect("pressed", self, "_on_button_pressed")
    _request.connect("request_completed", self, "_on_request_completed")


func _on_button_pressed():
    print("Doing request...")
    _request.set_download_file("the_thing.png")
    _request.request("http://docs.godotengine.org/en/stable/_static/docs_logo.png")


func _on_request_completed(result, response_code, headers, body):
    print("Request completed ", result, ", ", response_code)
```

The request completed with response 200 but with code 1 RESULT_CHUNKED_BODY_SIZE_MISMATCH.
And this error appeared in the console:

```
ERROR: DVector<unsigned char>::resize: Condition ' mem.is_locked() ' is true. returned: ERR_LOCKED
At: c:\projects\godot-builds\godot\core\dvector.h:365
```

Is this a bad use of the node or is it a bug?

The image file was successfully created, but was corrupted. It shows in some image viewers but not on the web, here is how it looks:

![the_thing_](https://cloud.githubusercontent.com/assets/1311555/21375062/4f951332-c72b-11e6-9e16-d6655c5d67b3.png)

Test project: 
[HttpRequest.zip](https://github.com/godotengine/godot/files/665302/HttpRequest.zip)

**Comment by volzhs** [comment]
possibly duplicated #6103

**Comment by akien-mga** [comment]
Assuming fixed (like #6103), please comment/open a new issue if it's not the case.
